### DISCUSSION TITLE
Indexed access via extension methods.

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [dotnet/roslyn#51735] [Ideas] [answered]
Can we please have the ability to overload square brackets in extension methods.

example...
Vector3[index]

A lot of math's code will use this notation, but we can't overload Numerics.Vector3 to make it compatible.

Another solution is to provide indexed access to Vector3 et al

**Comment by PathogenDavid** [comment, marked as answer]
> Can we please have the ability to overload square brackets in extension methods.

That would require a language change, which falls under [dotnet/csharplang](https://github.com/dotnet/csharplang/). There's already an existing proposal to add the ability to extend types beyond just methods: https://github.com/dotnet/csharplang/issues/1711

> Another solution is to provide indexed access to Vector3 et al

API proposals can be submitted to [dotnet/runtime](https://github.com/dotnet/runtime).
